Code/GraphMol/MolHash/hashfunctions.cpp

```diff
@@ -137,6 +144,8 @@ static unsigned int NMDetermineComponents(RWMol *mol, unsigned int *parts,
 
 static std::string NMMolecularFormula(RWMol *mol, const unsigned int *parts,
                                       unsigned int part) {
+  PRECONDITION(mol, "bad molecule");
+  PRECONDITION((!part || parts), "bad parts pointer");
```

Comment:
Just to be clear there is no parts[part==0] and part is 1 indexed?

Reply:
that's correct

Reply:
but I will add a test to make sure the molformula has works when the molecule has multiple fragments